jetty-servlet/src/main/java/org/eclipse/jetty/servlet/FilterHolder.java

```diff
@@ -221,6 +222,39 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
     }
 
+    /**
+     * Work out the class of the held {@link Filter} even before the {@link FilterHolder} has been started.
+     * @return the class of the held {@link Filter}, or null if not available.
+     */
+    @SuppressWarnings("unchecked")
+    public Class<? extends Filter> getFilterClass()
+    {
+        if (_filter != null)
+            return _filter.getClass();
+
+        Filter filter = getInstance();
+        if (filter != null)
+            return filter.getClass();
+
+        Class<? extends Filter> heldClass = getHeldClass();
+        if (heldClass != null)
+            return heldClass;
+
+        String className = getClassName();
+        if (className != null)
+        {
+            try
+            {
+                return Loader.loadClass(className);
+            }
+            catch (ClassNotFoundException e)
+            {
+                LOG.warn("Could not load filter class", e);
+            }
+        }
+        return null;
+    }
+
```

Comment:
I'm very cautious about this method!

firstly, this should probably be a mechanism on `BaseHolder` even if only used for filters.

Then if an instance has been set, then the heldClass will be set in the baseHolder, so you really can just look at `getHeldClass()` and only do something special if it is null. 

But my biggest concern is that the conversion of className to class is entirely dependent on the thread context classloader when called. This is well defined for when start is called, but not no define at all for when this method is called.... 

Need to think more about this... but please tidy up the above comments while I do!

Reply:
The other alternative is to use a very specific filter name.  Perhaps the fully qualified class name as the filter name?